atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/expectations/KeyValueServiceDataTracker.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl.expectations;
+
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.transaction.api.expectations.TransactionReadInfo;
+
+// todo(aalouane): implement this here
+public abstract class KeyValueServiceDataTracker {
+    public abstract TransactionReadInfo getReadInfo();
+
```

Comment:
can you introduce the interface in a separate PR

Reply:
This doesn't have to be an interface - I had it as such for the build to work

PR 1c is the implementation of this - I thought it was clearer to start with the `TrackingKeyValueServiceImpl` in this one (as some aspects of `KeyValueServiceDataTracker` don't look motivated without seeing the usage)

Still, I do agree this PR is too big - would have been a better choice if i had separated the utils/sizing functions from the actual kvs stuff